### PR TITLE
3912: Admin tool to unlock PDFs that are locked for editing

### PR DESCRIPTION
**Pull request by jamesgriff**

### CODE REVIEW

File: admin/tools/__init__.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    output_pdf_file.save(pdf_stream, deflate=True)
+    output_pdf_file.close()
+
+    pdf_stream.seek(0)
+    pdf_bytes = pdf_stream.read()
+    return pdf_bytes
+
+
+def get_filename_without_extension(input_file_name):
+    last_dot_in_input_filename = input_file_name.rindex('.')
```
Comment (alistairknight) [inline admin/tools/__init__.py:67]: This doesn't take into account files without an extension (it's a mac thing)

File: admin/tools/forms.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from flask_wtf import FlaskForm
+from wtforms import FileField, PasswordField
+from wtforms.validators import DataRequired
+
+
+class UnlockFileForm(FlaskForm):
+    file = FileField(
+        validators=[
```
Comment (alistairknight) [inline admin/tools/forms.py:8]: The file upload validator on the main site is more comprehensive e.g.:
`
        validators=[
            StrictRequiredIf('button_clicked', 'Upload file',
                             message='Select a PDF file smaller than 10MB',
                             validators=[
                                 MultiFileAllowed(['pdf'],
                                                  message='Select a PDF file smaller than 10MB'),
                                 fileSizeLimit(10)
                             ]),
        ]
    )
`

File: admin/tools/__init__.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            logger.log(LogLevel.ERROR, f"File could not be converted. Error was {e}")
+            form.file.errors.append(f"The file could not be converted. The error was: {e}")
+
+    return render_template(
+        'tools/unlock-file-locked-for-editing.html',
+        form=form
+    )
+
+
+def unlock_pdf(input_pdf_file):
```
Comment (alistairknight) [inline admin/tools/__init__.py:53]: The functions below are now fairly common throughout the site, I thing they would be better refactored into their own class

File: admin/tools/__init__.py (hunk 1 above)
Comment (jamesgriff) [reply admin/tools/__init__.py:67]: Good spot 👍
I'll make it work with or without a file extension

File: admin/tools/forms.py (hunk 2 above)
Comment (jamesgriff) [reply admin/tools/forms.py:8]: Yeh good idea, I'll add in those :-)

File: admin/tools/__init__.py (hunk 3 above)
Comment (jamesgriff) [reply admin/tools/__init__.py]: Good idea - I've moved all these methods into a separate "PDF Utils" file

**Review by alistairknight — APPROVED** [review]

**Pull request merged**